python/lsst/validate/drp/calcSrd.py

```diff
@@ -259,18 +267,225 @@ def computeWidths(array):
+                              visit_obj2, ra_obj2, dec_obj2):
+    """Match visit_obj1 and visit_obj2 and calculate ra, dec distance for matches.
+
+    For each visit shared between visit_obj1 and visit_obj2, calculate the spherical distance between the
+    matching elements in (ra_obj1, dec_obj1); (ra_obj2, dec_obj2)
+
+    Inputs
+    ------
+    visit_obj1 : scalar, list, or numpy.array of int or str
+        List of visits for object 1.
+    ra_obj1 : scalar, list, or numpy.array of float
+        List of RA for object 1.
+    dec_obj1 : scalar, list or numpy.array of float
+        List of Dec for object 1.
+    visit_obj2 : list or numpy.array of int or str
+        List of visits for object 2.
+    ra_obj2 : list or numpy.array of float
+        List of RA for object 2.
+    dec_obj2 : list or numpy.array of float
+        List of Dec for object 2.
+
+    Results
+    -------
+    list of float
+        spherical distances (in radians) for matching visits.
+    """
+    distances = []
+    for i in range(len(visit_obj1)):
+        for j in range(len(visit_obj2)):
+            if (visit_obj1[i] == visit_obj2[j]):
+                if np.isfinite([ra_obj1[i], dec_obj1[i],
+                                ra_obj2[j], dec_obj2[j]]).all():
+                    distances.append(sphDist(ra_obj1[i], dec_obj1[i],
+                                             ra_obj2[j], dec_obj2[j]))
+    return distances
+
+
+def arcminToRadians(arcmin):
+    return np.deg2rad(arcmin/60)
+
+def radiansToMilliarcsec(rad):
+    return np.rad2deg(rad)*3600*1000
+
+
+def calcAM1(*args, **kwargs):
+    """Calculate the SRD definition of astrometric performance for AM1
+
+    See `calcAMx` for more details."""
+    return calcAMx(*args, D=srdSpec.D1, width=2, **kwargs)
+
+def calcAM2(*args, **kwargs):
+    """Calculate the SRD definition of astrometric performance for AM2
+
+    See `calcAMx` for more details."""
+    return calcAMx(*args, D=srdSpec.D2, width=2, **kwargs)
+
+def calcAM3(*args, **kwargs):
+    """Calculate the SRD definition of astrometric performance for AM3
+
+    See `calcAMx` for more details."""
+    return calcAMx(*args, D=srdSpec.D3, width=2, **kwargs)
+
+def calcAMx(groupView, D=5, width=2, magrange=None):
+    """Calculate the SRD definition of astrometric performance
+
+    Parameters
+    ----------
+    groupView : lsst.afw.table.GroupView
+         GroupView object of matched observations from MultiMatch.
+    D : float
+        Fiducial distance between two objects to consider. [arcmin]
+    width : float
+        Width around fiducial distance to include. [arcmin]
+    magrange : 2-element list or tuple
+        brighter, fainter limits of the magnitude range to include.
+        E.g., `magrange=[17.5, 21.0]`
+
+    Returns
+    -------
+    list, list, list
+        rmsDistMas, annulus, magrange
+
+    Notes
+    -----
+    This table below is provided in this package in the `srdSpec.py` file.
```

Comment:
I hear you about not duplicating the numbers.  But how can I make it easily visible in the help for the function?

`srdSpec` is currently just a pipeBase.Struct.  I can't attach a docstring to it.